pkg/compiler/compilation_document.go

```diff
@@ -104,8 +112,16 @@ func postCompileHook(dir string, file core.File, hook string, log *zap.SugaredLo
 	}
 
 	cmd := exec.Command(hookSegments[0], args...)
-	log.Infof(`running post-output hook: %s`, strings.Join(cmd.Args, " "))
 	cmd.Dir = dir
+
+	quotedArgs := cmd.Args
+	for i, arg := range quotedArgs {
+		if !unquotedCharsRe.MatchString(arg) {
```

Comment:
This is user-visible at INFO log level, so imo it's decently important to have it look nice.

Instead of:

```
running post-output hook: npx prettier -w index.ts
```

it would be:

```
running post-output hook: "npx" "prettier" "-w" "index.ts"
```

... which I find harder to read, by enough to warrant the regex.

I'm a fan of xkcd as much as the next guy, but I don't think regex is really _that_ scary, especially at the level of complexity of this one.